plugins/vpc-eni/network/network_windows.go

```diff
@@ -198,6 +201,53 @@ func (nb *NetBuilder) FindOrCreateEndpoint(nw *Network, ep *Endpoint) error {
 		hnsEndpoint.PrefixLength = uint8(pl)
 	}
 
+	// Add ACL policies for blocking IMDS access through the endpoint.
+	if ep.BlockIMDS {
```

Comment:
These is a network.imds package in this repo that houses the same functionality for Linux. Do we want to create an imds_windows.go and put this block as a reusable function there? I'm undecided; this is a question.

Reply:
Yes that would be a good idea. That would ensure better functionality reuse in other plugin.
I'll make this code change.

Reply:
You probably want to have the blocking rule in IMDS package, and the "allow all" rules still in the plugin, as the exact policy can change from plugin to plugin.

Reply:
Allow rule is required to be a part of this entire ACL policy. If the block policy is applied without allow policy, all the traffic from the endpoint will be blocked.

The usage of other ACLs would require specific block/allow rules with higher/lower priority than the ones specified here.